codes/scripts/post_install.py

```diff
@@ -0,0 +1,142 @@
+import requests
+import zipfile
+if sys.version_info < (3, 8):
+    import importlib_metadata
+else:
+    import importlib.metadata as importlib_metadata
+
+req_file = os.path.join(os.getcwd(), "requirements.txt")
+
+def run(command, desc=None, errdesc=None, custom_env=None):
+    if desc is not None:
+        print(desc)
+
+    result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, env=os.environ if custom_env is None else custom_env)
+
+    if result.returncode != 0:
+
+        message = f"""{errdesc or 'Error running command'}.
+Command: {command}
+Error code: {result.returncode}
+stdout: {result.stdout.decode(encoding="utf8", errors="ignore") if len(result.stdout)>0 else '<empty>'}
+stderr: {result.stderr.decode(encoding="utf8", errors="ignore") if len(result.stderr)>0 else '<empty>'}
+"""
+        raise RuntimeError(message)
+
+    return result.stdout.decode(encoding="utf8", errors="ignore")
+
+def check_versions():
+    global req_file
+    reqs = open(req_file, 'r')
+    lines = reqs.readlines()
+    reqs_dict = {}
+    for line in lines:
+        splits = line.split("==")
+        if len(splits) == 2:
+            key = splits[0]
+            if "torch" not in key:
+                if "diffusers" in key:
+                    key = "diffusers"
+                reqs_dict[key] = splits[1].replace("\n", "").strip()
+    
+
+#uninstall tensorboard so it works
+#download dvae https://huggingface.co/jbetker/tortoise-tts-v2/resolve/3704aea61678e7e468a06d8eea121dba368a798e/.models/dvae.pth
+print("Downloading DVAE...")
+r = requests.get("https://huggingface.co/jbetker/tortoise-tts-v2/resolve/3704aea61678e7e468a06d8eea121dba368a798e/.models/dvae.pth", allow_redirects=True)
+#save to experiments
+open('experiments/dvae.pth', 'wb').write(r.content)
+
+print("Downloading Autoregressive Model...")
+r = requests.get("https://huggingface.co/jbetker/tortoise-tts-v2/resolve/main/.models/autoregressive.pth", allow_redirects=True)
+#save to experiments
+open('experiments/autoregressive.pth', 'wb').write(r.content)
+
+run("pip uninstall tensorboard -y", "Uninstalling Tensorboard")
+
+
+base_dir = os.path.dirname(os.getcwd())
+#repo = git.Repo(base_dir)
+#revision = repo.rev_parse("HEAD")
+#print(f"Dreambooth revision is {revision}")
+#check_versions()
+# Check for "different" B&B Files and copy only if necessary
+if os.name == "nt":
+    '''
+    python = sys.executable
+    bnb_src = os.path.join(os.getcwd(), "resources/bitsandbytes_windows")
+    bnb_dest = os.path.join(sysconfig.get_paths()["purelib"], "bitsandbytes")
+    '''
+    cudnn_src = os.path.join(os.getcwd(), "resources/cudnn_windows")
+    #check if chudnn is in cwd
+    if not os.path.exists(cudnn_src):
+        print("Can't find CUDNN in resources, trying main folder...")
+        cudnn_src = os.path.join(os.getcwd(), "cudnn_windows")
+        if not os.path.exists(cudnn_src):
+            cudnn_url = "https://b1.thefileditch.ch/mwxKTEtelILoIbMbruuM.zip"
```

Comment:
what the fuck is this

Reply:
I am not going to merge code that downloads executable code from an arbitrary filehosting site. Please find a more reputable source for CUDNN installation.

Reply:
I uploaded it there myself and it's the same link I use in my Stable tuner, if you have a website that you'd consider more reputable that can host the DLL files let me know, otherwise it can just be removed, it's meant to improve performance on 4000 cards.

Reply:
I will try to search for one

Reply:
well this is disappointing

it seems like the last public version of cudnn that is not locked behind a login is 8.3.0.98

I will have to think about whether I should just merge it

Reply:
@devilismyfriend I found the url!

https://developer.download.nvidia.com/compute/redist/cudnn/v8.6.0/local_installers/11.8/cudnn-windows-x86_64-8.6.0.163_cuda11-archive.zip

This downloads cudnn from nvidia without any login. You might want to take it for your own repo as well :)

Reply:
Nice, I'll updtlate the link and make sure everything is going to the right places